devdoc/two_dimensional_array_requirements.md

```diff
@@ -0,0 +1,106 @@
+# two_dimensional_array_requirements
+============================
+
+## Overview
+
+`two_dimensional_array` provides a 2D array with variable size of rows and columns. Each row has the same size to ensure easier access. The first dimention of array is resizable, it contains the pointer to the second dimention of array. The pointer is initiazlied to NULL and points to the allocated memory address when there's not enough space. When the memory stored in a row is not needed any more, the entire row shall be freed and set back to `NULL`.
```

Comment:
>The pointer is initiazlied to NULL and points to the allocated memory address when there's not enough space. When the memory stored in a row is not needed any more, the entire row shall be freed and set back to `NULL`. [](http://example.com/codeflow?start=237&length=218)

I think this part is starting to explain some of the design/implementation. It belongs to a different section, let's have one called design.

That section can detail even more.